include/core/platform/lf_atomic.h

```diff
@@ -0,0 +1,16 @@
+#ifndef LF_ATOMICS_H
+#define LF_ATOMICS_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+int32_t lf_atomic_fetch_add32(int32_t * ptr, int32_t val);
+int64_t lf_atomic_fetch_add64(int64_t * ptr, int64_t val);
+int32_t lf_atomic_add_fetch32(int32_t * ptr, int32_t val);
+int64_t lf_atomic_add_fetch64(int64_t * ptr, int64_t val);
+bool lf_atomic_bool_compare_and_swap32(int32_t* ptr, int32_t value, int32_t newval);
+bool lf_atomic_bool_compare_and_swap64(int64_t* ptr, int64_t value, int64_t newval);
+int32_t lf_atomic_val_compare_and_swap32(int32_t *ptr, int32_t val, int32_t newval);
+int64_t lf_atomic_val_compare_and_swap64(int64_t *ptr, int64_t val, int64_t newval);
```

Comment:
These functions all need descriptions.  I think that when constructing the descriptions, it may become clear whether they are correctly named or not.  As it stands, I have to look at the implementation to see what they mean. In particular, `lf_atomic_bool_compare_and_swap...` seem particularly strange because there is no `bool` involved.  What is compared?  What is swapped?  How is this different from `lf_atomic_val_compare_and_swap...`?

Reply:
Will add docs. They are modeled after the GCC atomics described here: https://gcc.gnu.org/onlinedocs/gcc/_005f_005fsync-Builtins.html

bool vs val is the return type from the compare and swap